src/otx/algo/instance_segmentation/segmentors/maskdino.py

```diff
@@ -0,0 +1,264 @@
+        """Predict function.
+
+        Args:
+            features (dict[str, Tensor]): feature maps
+            imgs_info (list[ImageInfo] | list[dict[str, Any]]):
+                list[ImageInfo]: image info (i.e ori_shape) list regarding original images used for training
+                list[dict[str, Any]]: image info (i.e img_shape) used for exporting
+            export (bool, optional): whether to export the model. Defaults to False.
+
+        Returns:
+            tuple[list[Tensor], list[torch.LongTensor], list[tv_tensors.Mask]]:
+                list[Tensor]: bounding boxes and scores with shape [N, 5]
+                list[torch.LongTensor]: labels with shape [N]
+                list[tv_tensors.Mask]: masks with shape [N, H, W]
+        """
+        outputs, _ = self(features)
+
+        class_queries_logits = outputs["pred_logits"]
+        masks_queries_logits = outputs["pred_masks"]
+        mask_box_results = outputs["pred_boxes"]
+
+        device = masks_queries_logits.device
+        num_classes = self.num_classes
+        num_queries = self.num_queries
+        test_topk_per_image = self.test_topk_per_image
+
+        batch_bboxes_scores: list[Tensor] = []
+        batch_labels: list[torch.LongTensor] = []
+        batch_masks: list[tv_tensors.Mask] = []
+
+        for mask_pred, mask_cls, pred_boxes, img_info in zip(
+            masks_queries_logits,
+            class_queries_logits,
+            mask_box_results,
+            imgs_info,
+        ):
+            h, w = img_info["img_shape"] if export else img_info.ori_shape  # type: ignore[index, attr-defined]
+            scores = mask_cls.sigmoid()
+            labels = torch.arange(num_classes, device=device).unsqueeze(0).repeat(num_queries, 1).flatten(0, 1)
+
+            scores_per_image, topk_indices = scores.flatten(0, 1).topk(test_topk_per_image, sorted=False)
+            labels_per_image = labels[topk_indices]
+
+            topk_indices = topk_indices // num_classes
+
+            mask_pred = mask_pred[topk_indices]  # noqa: PLW2901
+            pred_boxes = pred_boxes[topk_indices]  # noqa: PLW2901
+            pred_classes = labels_per_image
+
+            pred_masks = torch.nn.functional.interpolate(
+                mask_pred.unsqueeze(0),
+                size=(h, w),
+                mode="bilinear",
+                align_corners=False,
+            )[0]
+
+            pred_scores = scores_per_image * self.calculate_object_scores(pred_masks)
+            pred_boxes = pred_boxes.new_tensor([[w, h, w, h]]) * box_convert(  # noqa: PLW2901
+                pred_boxes,
+                in_fmt="cxcywh",
+                out_fmt="xyxy",
+            )
+            pred_boxes[:, 0::2].clamp_(min=0, max=w - 1)
+            pred_boxes[:, 1::2].clamp_(min=0, max=h - 1)
+
+            # Extract masks from RoI for exporting the model
+            if export:
+                pred_masks = self.roi_mask_extraction(pred_boxes, pred_masks) > 0
```

Comment:
```suggestion
                pred_masks = self.roi_mask_extraction(pred_boxes, pred_masks)
```
It seems that `roi_mask_extraction` already returns `cropped_masks > 0`. How about removing it?

Reply:
yes, you're right. I removed it :+1: